### PR TITLE
qa-pipelines: support embedded UAA

### PR DESCRIPTION
**Pull request by mook-as**
This is basically the same as the normal pipeline, but deploys UAA as part of the SCF deployment instead of via a separate helm chart.

### CODE REVIEW

File: qa-pipelines/tasks/cf-upgrade.sh [hunk 1]
```diff
@@ -52,33 +52,38 @@ set_uaa_sizing_params # Adds uaa sizing params to HELM_PARAMS
...
 # Deploy CF
-CA_CERT="$(get_internal_ca_cert)"
-
 set_helm_params # Resets HELM_PARAMS
 set_scf_sizing_params # Adds scf sizing params to HELM_PARAMS
 
 echo SCF customization ...
 echo "${HELM_PARAMS[@]}" | sed 's/kube\.registry\.password=[^[:space:]]*/kube.registry.password=<REDACTED>/g'
 
+if [ "${EMBEDDED_UAA:-false}" = "true" ]; then
```
Comment (prabalsharma) [inline qa-pipelines/tasks/cf-upgrade.sh]: why not [[ == ]]?

Comment (mook-as) [reply qa-pipelines/tasks/cf-upgrade.sh]: Fixed in bfa71748b534ee884718a00cb734a9dd21e3b621

File: qa-pipelines/tasks/cf-deploy.sh [hunk 2]
```diff
@@ -66,14 +66,19 @@ if [[ ${PROVISIONER} == kubernetes.io/rbd ]]; then
     kubectl get secret -o yaml ceph-secret-admin | sed "s/namespace: default/namespace: ${CF_NAMESPACE}/g" | kubectl create -f -
 fi
 
+if [[ "${EMBEDDED_UAA:-false}" == "true" ]]; then
+    HELM_PARAMS+=(--set "enable.uaa=true")
```
Comment (prabalsharma) [inline qa-pipelines/tasks/cf-deploy.sh]: you did this in qa-pipelines/tasks/lib/cf-deploy-upgrade-common.sh as well
https://github.com/SUSE/cf-ci/pull/270/files#diff-c47082a429cd24ef28f9c5f2a43d896fR191

File: qa-pipelines/tasks/lib/cf-deploy-upgrade-common.sh [hunk 3]
```diff
@@ -242,19 +245,20 @@ set -o allexport
...
+if [[ -n "${public_ip}" ]]; then
+    # If we have a public IP in the config map, we assume this is openstack / bare metal / etc. and have
+    # external IPs hard-coded.
     # Domain for SCF. DNS for *.DOMAIN must point to the same kube node
     # referenced by external_ip.
     DOMAIN=${public_ip}.${MAGIC_DNS_SERVICE}
+    # We use external_ips in set_helm_params()
+    external_ips=($(kubectl get nodes -o json | jq -r '.items[].status.addresses[] | select(.type == "InternalIP").address'))
+else
+    # If we do _not_ have a public IP, assume this is in the cloud™ somewhere and we will use Azure DNS
```
Comment (prabalsharma) [inline qa-pipelines/tasks/lib/cf-deploy-upgrade-common.sh]: lol, whats with TM

File: qa-pipelines/tasks/cf-deploy.sh (hunk 2 above)
Comment (mook-as) [reply qa-pipelines/tasks/cf-deploy.sh]: So I did! Fixed in a4822979141bf4d5269f5b1e37ee73f996e67766.

File: qa-pipelines/tasks/cf-deploy.sh [hunk 4]
```diff
@@ -30,31 +30,31 @@ fi
 echo UAA customization ...
 echo "${HELM_PARAMS[@]}" | sed 's/kube\.registry\.password=[^[:space:]]*/kube.registry.password=<REDACTED>/g'
 
-# Deploy UAA
-kubectl create namespace "${UAA_NAMESPACE}"
-if [[ "${PROVISIONER}" == "kubernetes.io/rbd" ]]; then
-    kubectl get secret -o yaml ceph-secret-admin | sed "s/namespace: default/namespace: ${UAA_NAMESPACE}/g" | kubectl create -f -
-fi
+if [ "${EMBEDDED_UAA:-false}" != "true" ]; then
```
Comment (prabalsharma) [inline qa-pipelines/tasks/cf-deploy.sh]: [[ ]] for consistency

File: qa-pipelines/tasks/cf-upgrade.sh [hunk 5]
```diff
@@ -52,33 +52,36 @@ set_uaa_sizing_params # Adds uaa sizing params to HELM_PARAMS
...
-helm upgrade uaa ${CAP_DIRECTORY}/helm/uaa/ \
-    --namespace "${UAA_NAMESPACE}" \
-    --recreate-pods \
-    --timeout 3600 \
-    --wait \
-    "${HELM_PARAMS[@]}"
-
-# Wait for UAA release
-wait_for_release uaa
+if [ "${EMBEDDED_UAA:-false}" != "true" ]; then
```
Comment (prabalsharma) [inline qa-pipelines/tasks/cf-upgrade.sh]: [[ ]] for consistency

File: qa-pipelines/tasks/lib/cf-deploy-upgrade-common.sh [hunk 6]
```diff
@@ -187,6 +187,9 @@ set_helm_params() {
             HELM_PARAMS+=(--set "kube.external_ips[$i]=${external_ips[$i]}")
         done
     fi
+    if [ "${EMBEDDED_UAA:-false}" = "true" ]; then
```
Comment (prabalsharma) [inline qa-pipelines/tasks/lib/cf-deploy-upgrade-common.sh]: [[]] for consistency

File: qa-pipelines/tasks/cf-deploy.sh (hunk 4 above)
Comment (mook-as) [reply qa-pipelines/tasks/cf-deploy.sh]: Fixed in 86a2d8d13026096aad7fecfcb1d52bb81fe4f16e.

File: qa-pipelines/tasks/cf-upgrade.sh (hunk 5 above)
Comment (mook-as) [reply qa-pipelines/tasks/cf-upgrade.sh]: Fixed in 86a2d8d13026096aad7fecfcb1d52bb81fe4f16e.

File: qa-pipelines/tasks/lib/cf-deploy-upgrade-common.sh (hunk 6 above)
Comment (mook-as) [reply qa-pipelines/tasks/lib/cf-deploy-upgrade-common.sh]: Fixed in 86a2d8d13026096aad7fecfcb1d52bb81fe4f16e along with the following `if`s.

**Pull request merged**